### PR TITLE
Adding API key ID and API key secret

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/20119

## Description
Updated the docker version to include the updated resilient client library
Added an option to authenticate using api key id and api key secret

### CODE REVIEW

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py [hunk 1]
```diff
@@ -994,13 +996,32 @@ def test():
...
+        resilient_client = resilient.get_client({
+            'host': SERVER,
+            'cafile': 'true' if USE_SSL else 'false',
+            'org': ORG_NAME,
+            'api_key_id': API_KEY_ID,
+            'api_key_secret': API_KEY_SECRET
+        })
+    else:
+        return_error('Credentials were not provided, please configure the username and password'
+                     ' or the api_key_id and api_secret_id')
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py:1020]:
```suggestion
                     ' or the API Key and API Secret')
```

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py [hunk 2]
```diff
@@ -994,13 +996,32 @@ def test():
...
+    elif API_KEY_ID and API_KEY_SECRET:
+        resilient_client = resilient.get_client({
+            'host': SERVER,
+            'cafile': 'true' if USE_SSL else 'false',
+            'org': ORG_NAME,
+            'api_key_id': API_KEY_ID,
+            'api_key_secret': API_KEY_SECRET
+        })
+    else:
+        return_error('Credentials were not provided, please configure the username and password'
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py:1019]:
```suggestion
        return_error('Credentials were not provided. Configure the username and password'
```

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py [hunk 3]
```diff
@@ -994,13 +996,32 @@ def test():
...
-    'password': PASSWORD,
-    'host': SERVER,
-    'cafile': 'true' if USE_SSL else 'false',
-    'org': ORG_NAME
-})
+
+
+def get_client():
+    if USERNAME and PASSWORD:
+        resilient_client = resilient.get_client({
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py:1003]: id prefer we init the client once, so lets build the args dict, assign it in the conditions and the pass it with the `**` operator

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/README.md [hunk 4]
```diff
@@ -12,7 +12,7 @@
 <ul>
 <li><strong>Name</strong>: a textual name for the integration instance</li>
 <li><strong>Server URL</strong></li>
-<li><strong>Credentials</strong></li>
+<li><strong>Credentials (either username and password or API key ID and API key secret)</strong></li>
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/README.md:15]: need to add to the readme and the details description how to get the api key and api secret

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py [hunk 5]
```diff
@@ -994,13 +996,32 @@ def test():
...
+    elif API_KEY_ID and API_KEY_SECRET:
+        resilient_client = resilient.get_client({
+            'host': SERVER,
+            'cafile': 'true' if USE_SSL else 'false',
+            'org': ORG_NAME,
+            'api_key_id': API_KEY_ID,
+            'api_key_secret': API_KEY_SECRET
+        })
+    else:
+        return_error('Credentials were not provided. Configure the username and password'
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py:1019]:
```suggestion
        return_error('Credentials were not provided. Configure either the username and password'
```

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py [hunk 6]
```diff
@@ -994,13 +996,33 @@ def test():
...
-    'host': SERVER,
-    'cafile': 'true' if USE_SSL else 'false',
-    'org': ORG_NAME
-})
+
+
+def get_client():
+    if USERNAME and PASSWORD:
+        opts_dict = {
+            'email': USERNAME,
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems.py:1004]: lets init `opts_dict` before the condition with the common keys (host, cafile, host and org) and then just set the creds

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/README.md (hunk 4 above)
Comment (hod-alpert) [reply Packs/IBMResilientSystems/Integrations/IBMResilientSystems/README.md:15]: I'll see where can I find these details

**Comment by kirbles19** [comment]
@hod-alpert Done.

**Comment by hod-alpert** [comment]
@bakatzir Since in line 1024, the method `get_client` checks connectivity and authorization and since I don't have credentials for this integration to check additional changes, I think I'll leave it with no further changes then what I've made.

File: Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems_description.md [hunk 7]
```diff
@@ -0,0 +1,13 @@
+
+## Configure the IBM Resilient Systems Integration on Demisto
+
+1.  Click **Add instance** to create and configure a new integration instance.
+    *   **Name**: a textual name for the integration instance
```
Comment (Itay4) [inline Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems_description.md:5]: no need for all of that here, only the comment about the credentials

**Comment by hod-alpert** [comment]
@kirbles19 Can you go over [this](https://github.com/demisto/content/blob/b2afa3f6c02552b88f3cb79371cc40ba8c65cdd6/Packs/IBMResilientSystems/Integrations/IBMResilientSystems/IBMResilientSystems_description.md) file one last time?

**Pull request merged**